src/mscorlib/src/System/String.Manipulation.cs

```diff
@@ -584,6 +595,42 @@ public static string Join(string separator, params object[] values)
             return StringBuilderCache.GetStringAndRelease(result);
         }
 
+        // Joins an array of objects together as one string with a char separator between each original string.
+        //
+        [ComVisible(false)]
+        public static String Join(Char separator, params Object[] values) {
+            if (values == null)
+                throw new ArgumentNullException("values");
+
+            if (values.Length == 0 || values[0] == null)
```

Comment:
We don't want to propogate the bug where `string.Join(string, object[])` returns null if the first item is null (even if the object has some non-null items) to more places.